pandas/tests/series/test_io.py

```diff
@@ -37,18 +37,36 @@ def read_csv(self, path, **kwargs):
     def test_from_csv_deprecation(self):
         # see gh-17812
         with ensure_clean() as path:
-            self.ts.to_csv(path)
+            self.ts.to_csv(path, header=False)
 
             with tm.assert_produces_warning(FutureWarning,
                                             check_stacklevel=False):
                 ts = self.read_csv(path)
                 depr_ts = Series.from_csv(path)
                 assert_series_equal(depr_ts, ts)
 
+    @pytest.mark.parametrize("arg", ["path", "header", "both"])
+    def test_to_csv_deprecation(self, arg):
+        # see gh-19715
+        with ensure_clean() as path:
+            if arg == "path":
+                kwargs = dict(path=path, header=False)
+            elif arg == "header":
+                kwargs = dict(path_or_buf=path)
```

Comment:
Why would this case raise the warning? (as `path_or_buf` is the future keyword?)

Reply:
Because `header` will default to `True` in the future (instead of `False` as it was the case in `Series.to_csv`).  Thus, users who don't specify the `header` argument will be "surprised" when that change in the default results in different output.  Thus, we are warning when `header` is not passed in explicitly to let them know that the default is changing.